src/deploy/scope_resourcegroup.ts

```diff
@@ -56,21 +56,24 @@ export async function DeployResourceGroupScope(azPath: string, validationOnly: b
     // validate the deployment
     core.info("Validating template...")
     var code = await exec(`"${azPath}" deployment group validate ${azDeployParameters} -o json`, [], validateOptions);
-    if (validationOnly && code != 0) {
+    if (deploymentMode.toLowerCase() === "validate" && code != 0) {
         throw new Error("Template validation failed.")
     } else if (code != 0) {
         core.warning("Template validation failed.")
     }
 
-    // execute the deployment
-    core.info("Creating deployment...")
-    var deploymentCode = await exec(`"${azPath}" deployment group create ${azDeployParameters} -o json`, [], deployOptions);
-    if (deploymentCode != 0) {
-        core.error("Deployment failed.")
-    }
-    core.debug(commandOutput);
+    if (deploymentMode.toLowerCase() != "validate") {
+        // execute the deployment
+        core.info("Creating deployment...")
+        var deploymentCode = await exec(`"${azPath}" deployment group create ${azDeployParameters} -o json`, [], deployOptions);
```

Comment:
I think we need to update the `--mode` argument in `azDeployParameters` to be incremental rather than validate, as validate is not a valid mode for the az command

Reply:
should it not have failed due to this?

Reply:
Also, if I am right, it need not be done for subscription and mg level deployments as they don't have `--mode` argument

Reply:
yes, only for resource group scenario